tianshou/data/batch.py

```diff
@@ -29,6 +29,43 @@ def _is_batch_set(data: Any) -> bool:
     return False
 
 
+def _is_scalar(value: Any) -> bool:
+    # check if the value is a scalar
+    # 1. python bool object, number object: isinstance(value, Number)
+    # 2. numpy scalar: isinstance(value, np.generic)
+    # 3. python object rather than dict / Batch / tensor
+    # the check of dict / Batch is omitted because this only checks a value.
+    # a dict / Batch will eventually check their values
+    value = np.asanyarray(value)
+    return value.size == 1 and not value.shape
```

Comment:
Maybe your previous implementation was more appropriate after all, if you need to specify `and not value.shape`. I don't know.